src/agoradatatools/great_expectations/gx/plugins/expectations/expect_column_nested_field_values_to_be_mostly_not_null.py

```diff
@@ -0,0 +1,329 @@
+
+        # parse json in the column
+        series_parsed = column.apply(cls.safe_parse)
+
+        counts = cls._flatten_nested_object_count_nulls(
+            list_object=list(series_parsed), target_field=target_field
+        )
+
+        if counts["total_dict"]:
+            return round(
+                (counts["total_dict"] - counts["total_nulls"]) / counts["total_dict"], 2
+            )
+        else:
+            raise ValueError("There are no JSON objects to validate.")
+
+    @staticmethod
+    def _flatten_nested_object_count_nulls(
+        list_object: list[list[dict[str, str | int | bool | None]]], target_field: str
+    ) -> dict[str, int]:
+        """
+        Recursively flattens a nested list of dictionaries and counts how many
+        dictionaries have a null value for the specified target field.
+
+        Example:
+        list_object = [
+            '[{"target_field": "value1", "other_key": "info"}, {"target_field": "value2", "other_key": "info"}]', //all valid
+            '[{"target_field": null}, {"target_field": "value2", "other_key": "info"}]', // one INVALID: contain null
+            '[{"target_field": ""}]', // one valid: contain an empty string
+            '[{"another_key": ""}]', // one INVALID: target_field is missing
+            '[{}]', // one INVALID: target_field is missing
+            '[]' //IGNORE
+            'null' //IGNORE
+        ]
+
+        The code recursively counts nulls for a field in nested dict lists.
+        total_nulls = 3
+        total_nulls = 7
+
+        Parameters:
+            list_object (list of dictionaries): A potentially nested list containing dictionaries.
+            target_field (str): The key to check for null values within each dictionary.
+
+        Returns:
+            dictionary that contains two keys:
+                - total_nulls (int): The number of dictionaries where target_field is None.
+                - total_dict (int): The total number of dictionaries encountered that contain the target_field.
+        Note:
+        A value is counted as invalid if:
+            - The target field is missing from the dictionary
+            - The value is None
+        Empty lists are ignored entirely.
+        Empty string is considered as valid.
+        """
+        counts = {"total_nulls": 0, "total_dict": 0}
+
+        def _flatten(list_object: list[str]) -> dict[str, int]:
+            """
+            Recursively flattens a nested list of dictionaries and counts how many
+            dictionaries have a null value for the specified target field.
+            """
+            # if this only contains empty list
+            for item in list_object:
+                if isinstance(item, list):
+                    _flatten(item)
+                elif isinstance(item, dict):
+                    if target_field not in item or item.get(target_field) is None:
+                        counts["total_nulls"] += 1
+                    counts["total_dict"] += 1
```

Comment:
`total_dict` is _The total number of dictionaries encountered that contain the target_field_, but this logic would make any `dict` item contribute to the `total_dict` count. I would refactor this as:

```
if target_field in item:
  counts["total_dict"] += 1
  if item[target_field] is None:
    counts["total_nulls"] += 1
else:
  counts["total_nulls"] += 1
```